### DISCUSSION TITLE
Cart Lines Update sets quantity to 0 when attempting to update a line item above the available inventory

### DISCUSSION THREAD

**Discussion by ndimatteo** [Shopify/storefront-api-feedback#186] [Ideas & Requests]
When updating cart lines with a new quantity _**above**_ the available inventory set in Shopify results in the API returning a quantity of `0` and leaving the item in the cart.

This only happens on line items that **are not** the first line item in the cart for some reason.

This is [reproducible with the`hyrdogen-react` lib](https://github.com/Shopify/hydrogen/issues/824), which I was redirected by @frehner to post about here.

**Comment by johncraigcole** [comment]
Hey @ndimatteo thanks for sharing those details. An issue has been created on our end but I don't have a timeline I can share on when the issue will be resolved. I'll follow up here when I know more.

**Comment by ndimatteo** [comment]
Thanks @johncraigcole! Appreciate the quick response 🤘

**Comment by steve-jiang** [comment]
It is also not consistent, sometimes it sets the quantity to 0 and sometimes it just fails silently. 

This is especially bad when the Cart SFAPI is used in conjunction with Shopify Script because the Script executes after the Cart Mutation and the Script doesn't have the ability to check if a line item is out of stock or not. The script will then consider the line quantity as set by the Cart mutation and produce unpredictable results by honouring discounts when it shouldn't.
